src/player.rs

```diff
@@ -70,3 +81,39 @@ pub fn spawn_player(mut commands: Commands, asset_server: Res<AssetServer>) {
         },
     ));
 }
+
+// This is my first attempt at a laser firing system, for now I just want it to work, but in the future I hope to implement rapier, and make this far more interesting if possible.
+
+pub fn spawn_laser(
+    mut commands: Commands,
+    input: Res<Input<KeyCode>>,
+    query: Query<&Transform, With<Player>>,
+) {
+    if input.just_pressed(KeyCode::Space) {
+        if let Ok(player_transform) = query.get_single() {
+            commands
+                .spawn(SpriteBundle {
+                    sprite: Sprite {
+                        color: Color::RED,
+                        custom_size: Some(Vec2::new(5.0, 15.0)),
+                        ..default()
+                    },
+                    transform: Transform {
+                        translation: player_transform.translation + Vec3::Y * 20.0,
+                        ..default()
+                    },
+                    ..default()
+                })
+                .insert(Laser)
+                .insert(Velocity {
+                    linvel: Vec2::new(0.0, LASER_SPEED),
+                });
+        }
+    }
+}
+
+fn velocity(time: Res<Time>, mut query: Query<(&Velocity, &mut Transform)>) {
```

Comment:
Is this something that might be better placed in physics instead of here?

I thought about it when I realized we could just have velocity as a field in laser, but actually this is more general and would affect anything with Velocity, which could include the player, lasers, enemies, etc. 

Though this might need some refactoring on the Krill side too. Just food for thought, not a blocker.